src/components/Avatar/Avatar.scss

```diff
@@ -0,0 +1,159 @@
+@import "../../styles/themes.scss";
+@import "../../styles/typography.scss";
+@import "../../styles/states.scss";
+
+.monday-style-avatar {
+  position: relative;
+  &_circle {
+    position: relative;
+    border: 1px solid;
+    border-radius: 50%;
+    overflow: hidden;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    @include theme-prop(color, text-color-on-primary);
+
+    &:focus {
+      outline: none;
+      @include focus-style(50%);
+    }
+
+    &--large {
+      $avatar-size-large: 50px;
+      height: $avatar-size-large;
+      width: $avatar-size-large;
+
+      .monday-style-avatar_text {
+        font-size: $font-size-avatar-large;
+      }
+
+      .monday-style-avatar_icon {
+        $avatar-icon-large: 28px;
+        height:  $avatar-icon-large;
+        width: $avatar-icon-large;
+      }
+
+    }
+
+    &--medium {
+      $avatar-size-medium: 36px;
+      height: $avatar-size-medium;
+      width: $avatar-size-medium;
+
+      .monday-style-avatar_text {
+        font-size: $font-size-avatar-medium;
+      }
+
+      .monday-style-avatar_icon {
+        $avatar-icon-medium: 18px;
+        height:  $avatar-icon-medium;
+        width: $avatar-icon-medium;
+      }
+
+    }
+
+    &--small {
+      $avatar-size-small: 28px;
+      height: $avatar-size-small;
+      width: $avatar-size-small;
+
+      .monday-style-avatar_text {
+        font-size: $font-size-avatar-small;
+      }
+
+      .monday-style-avatar_icon {
+        $avatar-icon-small: 12px;
+        height:  $avatar-icon-small;
+        width: $avatar-icon-small;
+      }
+    }
+
+    &--img {
+      @include theme-prop(border-color, primary-background-color);
+    }
+
+    &--text {
+      @include theme-prop(border-color, layout-border-color);
+    }
+
+    &--is-disabled {
+      &:before {
+        content: "";
+        height: 100%;
+        width: 100%;
+        position: absolute;
+        top: 0;
+        opacity: 0.5;
+        @include theme-prop(background, disabled-background-color);
+      }
+    }
+
+    &--is-square {
+      border-radius: $border-radius-small;
+      &:focus {
+        outline: none;
+        @include focus-style($border-radius-small);
+      }
+    }
+  }
+
+  &_image {
+    object-fit: cover;
```

Comment:
in the end I used object fit which seems to be supported on most of the browsers: https://caniuse.com/object-fit
am I still need to add css fallback? if so, how this is related to media queries? (I did not understand)

Reply:
no it is ok